twyla/service/test/unit/test_message.py

```diff
@@ -0,0 +1,101 @@
+            "message_type": "integration-request",
+            "bot_slug": "slow-slug",
+            "content": {},
+            "channel": "fbmessenger",
+            "channel_user_id": "some-user-id"
+        }
+        '''
+        self.invalid_test_body = '''
+        {
+            "message_type": "integration-request",
+            "bot_slug": "slow-slug",
+            "content": {},
+            "channel": "fbmessenger",
+            "channel_user_id": "some-user-id"
+        }
+        '''
+
+    def test_payload_parsing(self):
+        payload = message.EventPayload.parse_raw(self.test_body)
+
+        assert isinstance(payload.meta, message.Meta)
+        assert isinstance(payload.content, message.Content)
+        assert payload.tenant == 'test-tenant'
+        assert payload.message_type == 'integration-request'
+        assert payload.bot_slug == 'slow-slug'
+        assert payload.channel == 'fbmessenger'
+        assert payload.channel_user_id == 'some-user-id'
+
+
+    def test_payload_serialization_roundtrip(self):
+        payload = message.EventPayload.parse_raw(self.test_body)
+        raw_json = payload.to_json()
+
+        new_payload = message.EventPayload.parse_raw(raw_json)
+
+        assert payload.meta.timestamp == new_payload.meta.timestamp
+        assert payload.meta.session_id == new_payload.meta.session_id
+        assert payload.message_type == new_payload.message_type
+        assert payload.bot_slug == new_payload.bot_slug
+        assert payload.channel == new_payload.channel
+        assert payload.channel_user_id == new_payload.channel_user_id
+
+
+    def test_event_class(self):
+        mock_envelope = mock.MagicMock()
+        mock_envelope.delivery_tag = 1
+        mock_channel = helpers.AsyncMock()
+        event = message.Event(
+            channel=mock_channel,
+            body=self.test_body,
+            envelope=mock_envelope,
+            name='get_booking')
+
+        payload = helpers.aio_run(event.payload())
+
+        assert isinstance(payload, message.EventPayload)
+
+        helpers.aio_run(event.ack())
+        event.channel.basic_client_ack.assert_called_once_with(
+            mock_channel,
+            delivery_tag=1)
+
+        helpers.aio_run(event.reject())
+        event.channel.basic_reject.assert_called_with(
+            mock_channel,
+            delivery_tag=1,
+            requeue=True)
+
+        helpers.aio_run(event.drop())
+        event.channel.basic_reject.assert_called_with(
+            mock_channel,
+            delivery_tag=1,
+            requeue=False)
+
+        assert event.channel.basic_reject.call_count == 2
+
+
+    def test_event_class_bad_body(self):
+        event = message.Event(
+            channel=helpers.AsyncMock(),
+            body=self.invalid_test_body,
+            envelope=mock.MagicMock(),
+            name='get_booking')
+
+        with pytest.raises(pydantic.ValidationError):
+            _ = helpers.aio_run(event.payload())
```

Comment:
That is what I thought but actually some async parts work differently when assigned.. (I don't really know why, it would probably make sense to figure that out).

Here is some example.
```
import asyncio

async def raise_me(assigned=False):
    if assigned:
        print('Look how I do not visibly raise, yet!')
    raise RuntimeError()

loop = asyncio.get_event_loop()
loop.create_task(raise_me())
fut = loop.create_task(raise_me(assigned=True))

print('kill me with Ctrl-C')
loop.run_forever()
```

Reply:
So if the `future` is referenced it behaves differently

Reply:
I checked and in this case, the result is the same in both cases :D